docs/architecture.rst

```diff
@@ -1,142 +1,211 @@
+            "my_object", issue an instruction for its worker to call the method "my_method" on it, and
+            return the result.
+        d. The HTTP server and workers can handle thousands of concurrent calls per second, and have similar latency
+            under simple conditions to Flask.
+        e. New workers can be constructed with `rh.env`, which specifies the details of the Python environment
+            (packages, environment variables) in which the process will be constructed. By default workers live
+            in the same Python environment as the daemon, but can also be started in a conda environment or a
+            separate node.
+3. **Deploying Functions or Classes**: The user specifies a function or class to be deployed to the remote compute
+    using the `rh.function` or `rh.module` constructors (or by subclassing `rh.Module`), and calling
+    `remote_obj = my_obj.to(my_cluster, worker=my_env)`. The Runhouse client library extracts the path, module name,
+    and importable name from the function or class. If the function or class is defined in local code, the repo or
+    package is rsynced onto the cluster. An instruction with the import path is sent to the cluster to
+    construct the function or class in a particular worker and assign it a name in the key-value store, perhaps
+    overwriting an existing object.
+4. **Calling the Function or Class**: After deploying the function, class, or object into the server, the Runhouse
+    Python client returns a local callable stub which behaves like the original object, but forwards method calls
+    over HTTP to the remote object on the cluster.
+        a. If a stateful instance of a class is desired, an `__init__` method can be called on the remote class to
+            instantiate a new remote object from the class and assign it a name.
+        b. If arguments are passed to the method, they're serialized with cloudpickle and sent with the HTTP request.
+            Serializing code, such as functions, classes, or dataclasses, is strongly discouraged, as it can lead to
+            versioning mismatch errors between local and remote package versions.
+        c. From here on, you can think of Runhouse as facilitating
+            regular object-oriented programming but with the objects living remotely, maybe in a different cluster,
+            region, or cloud than the local code.
+        d. Python behavior like async, exceptions, printing, and logging are all preserved across remote calls, but
+            can be disabled or controlled if desired.
+5. **Saving and Loading**: The Runhouse client can save and load objects to and from the local filesystem, or to a
+    remote metadata store. This allows for easy sharing of clusters and services across users and environments,
+    and for versioning and rollback of objects. The metadata store can be accessed from any Python interpreter,
+    and is backed by a management API to view and manage all resources.
```

Comment:
i know we had this before, but "management API" makes it sound more like REST API reference docs, rather than something like a "monitoring or management portal"